test-runner/src/test.ts

```diff
@@ -156,4 +158,118 @@ export class Suite {
     });
     return found;
   }
+
+  async run(options: {
```

Comment:
- we can't expose `run` on the test/suite since test/suite are a public API
- i'd prefer to keep test/suite declarative & shallow classes, external runner pattern allows interchangeable runners with the same test descriptor and the specs, hence a better design.